rpc/helpers.go

```diff
@@ -60,22 +60,34 @@ func (h *Handler) blockByID(id *BlockID) (*core.Block, *jsonrpc.Error) {
 	return block, nil
 }
 
-func (h *Handler) blockHeaderByID(id *BlockID) (*core.Header, *jsonrpc.Error) {
+func (h *Handler) blockHeaderByID(id blockID) (*core.Header, *jsonrpc.Error) {
 	var header *core.Header
 	var err error
-	switch {
-	case id.Latest:
-		header, err = h.bcReader.HeadsHeader()
-	case id.Hash != nil:
-		header, err = h.bcReader.BlockHeaderByHash(id.Hash)
-	case id.Pending:
-		var pending *sync.Pending
-		pending, err = h.syncReader.Pending()
-		if err == nil {
-			header = pending.Block.Header
+	switch id := id.(type) {
+	case *BlockID:
```

Comment:
It is confusing having a struct named `BlockID` and an interface named `blockID`. 
Also, this function has some repeated code.

Proposal: let's change the function to be generic on the interface `blockID` (using the same name, but ideally, it will have another less confusing, perhaps `subscriptionID`)
```go
interface blockID {
  // methods to get `Latest`, `Hash` and `Pending`
}
```
Have `BlockID` and `SubscriptionBlockID` implement the interface.